test/benchmark/l4bench.go

```diff
@@ -0,0 +1,194 @@
+				res = vm.ContainerExec(helpers.Client, cmd)
+			} else {
+				res = vm.ContainerExec(helpers.Server, cmd)
+			}
+
+			res.ExpectSuccess()
+			fmt.Fprintf(&PerfLogWriter, "%s,", strings.TrimSuffix(res.GetStdOut(), "\n"))
+		}
+
+		SuperNetperfRR := func(client bool, num int, serverIP string, serverIPv6 string) {
+			SuperNetperfRRIPv4(client, num, serverIP)
+			ClientServerCheck(serverIPv6, serverIP)
+		}
+
+		SuperNetperfStreamIPv4 := func(client bool, num int, serverIP string) {
+			var res *helpers.CmdRes
+
+			By("super_netperf to %s from %s (should succeed)", helpers.Server, helpers.Client)
+			cmd := fmt.Sprintf("super_netperf %d -f g -t TCP_STREAM -H %s", num, serverIP)
+			if client {
+				res = vm.ContainerExec(helpers.Client, cmd)
+			} else {
+				res = vm.ContainerExec(helpers.Server, cmd)
+			}
+
+			res.ExpectSuccess()
+			fmt.Fprintf(&PerfLogWriter, "%s,", strings.TrimSuffix(res.GetStdOut(), "\n"))
+		}
+
+		SuperNetperfStream := func(client bool, num int, serverIP string, serverIPv6 string) {
+			SuperNetperfStreamIPv4(client, num, serverIP)
+			ClientServerCheck(serverIPv6, serverIP)
+		}
+
+		NetperfTest := func() {
+			serverData := vm.ContainerInspect(helpers.Server)
+			_serverIP, err := serverData.Filter(fmt.Sprintf("{[0].NetworkSettings.Networks.%s.IPAddress}", helpers.CiliumDockerNetwork))
+			Expect(err).Should(BeNil())
+
+			_serverIPv6, err := serverData.Filter(fmt.Sprintf("{[0].NetworkSettings.Networks.%s.GlobalIPv6Address}", helpers.CiliumDockerNetwork))
+			Expect(err).Should(BeNil())
+
+			serverIP := _serverIP.String()
+			serverIPv6 := _serverIPv6.String()
+			By("serverIP: %s", serverIP)
+			By("serverIPv6: %s", serverIPv6)
+
+			fmt.Fprintf(&PerfLogWriter, "lo netperf req/sec: ")
+			SuperNetperfRR(false, 1, serverIP, serverIPv6)
+			SuperNetperfRR(false, 10, serverIP, serverIPv6)
+			SuperNetperfRR(false, 100, serverIP, serverIPv6)
+			SuperNetperfRR(false, 1000, serverIP, serverIPv6)
+			fmt.Fprintf(&PerfLogWriter, "\ninter-container netperf req/sec: ")
+			SuperNetperfRR(true, 1, serverIP, serverIPv6)
+			SuperNetperfRR(true, 10, serverIP, serverIPv6)
+			SuperNetperfRR(true, 100, serverIP, serverIPv6)
+			SuperNetperfRR(true, 1000, serverIP, serverIPv6)
+			fmt.Fprintf(&PerfLogWriter, "\nlo netperf stream: ")
+			SuperNetperfStream(false, 1, serverIP, serverIPv6)
+			SuperNetperfStream(false, 10, serverIP, serverIPv6)
+			SuperNetperfStream(false, 100, serverIP, serverIPv6)
+			SuperNetperfStream(false, 1000, serverIP, serverIPv6)
+			fmt.Fprintf(&PerfLogWriter, "\ninter-container netperf stream: ")
+			SuperNetperfStream(true, 1, serverIP, serverIPv6)
+			SuperNetperfStream(true, 10, serverIP, serverIPv6)
+			SuperNetperfStream(true, 100, serverIP, serverIPv6)
+			SuperNetperfStream(true, 1000, serverIP, serverIPv6)
+			fmt.Fprintf(&PerfLogWriter, "\n")
+		}
+
+		It("Test L4 Netperf Performance", func() {
+			NetperfTest()
+		}, 300)
+
+		It("Test L4 Netperf Performance with Sockmap", func() {
```

Comment:
To do what I stated above by separating the tests into individual `It`, you could put this in a separate context called `SockMap Netperf`